docs/CppInteroperability/GettingStartedWithC++Interop.md

```diff
@@ -4,55 +4,46 @@ This document is designed to get you started with bidirectional API-level intero
 
 ## Table of Contents
 
+- [Installing Toolchain](#installing-toolchain)
 - [Creating a Module to contain your C++ source code](#creating-a-module-to-contain-your-c-source-code)
 - [Adding C++ to an Xcode project](#adding-c-to-an-xcode-project)
 - [Creating a Swift Package](#Creating-a-Swift-Package)
 - [Building with CMake](#building-with-cmake)
+- [Common Errors](#common-errors)
 
-## Creating a Module to contain your C++ source code
-
-- Create a new target in Xcode via _File_ | _New_ | _Target_, select _Library_.
-- Within the directory of the newly created target, create a new C++ implementation and header file
-- For this example we will call the files CxxTest, so we should have a CxxTest.cpp and CxxTest.hpp.
-- Next create an empty file and call it `module.modulemap`, in this file create the module for your source code, and define your C++ header (`requires cplusplus` isn't required but it's convention for C++ modules, especially if they use C++ features).
+## Installing Toolchain 
 
-```
-// In module.modulemap
-module CxxTest {
-    header "CxxTest.hpp"
-    requires cplusplus
-}
-```
+For ensuring the availability of the features we used in this demo. Please install the latest version of the toolchain from "Development Snapshots" on [swift.org](https://www.swift.org/download/#snapshots).
 
-<img width="127" alt="cxx-interop-ctest" src="https://user-images.githubusercontent.com/3801618/192995602-f37137f3-ec15-4fdd-bf2c-591728945a68.png">
+After finishing installing, go to the Xcode menu, select Xcode | Toolchains, and pick the latest toolchain.
 
-## Adding C++ to an Xcode project
-- In your xcode project, follow the steps [Creating a Module to contain your C++ source code](#creating-a-module-to-contain-your-c-source-code) in your project directory
+## Creating a Module to contain your C++ source code
 
-Add the C++ module to the include path and enable C++ interop:
-- Navigate to your project directory
-- In `Project` navigate to `Build Settings` -> `Swift Compiler`
-- Under `Custom Flags` -> `Other Swift Flags` add`-enable-experimental-cxx-interop`
-- Under `Search Paths` -> `Import Paths` add your search path to the C++ module (i.e, `./ProjectName/CxxTest`). Repeat this step in `Other Swift Flags`
+- Create a new Xcode project.
+- Create a new target in Xcode via _File_ | _New_ | _Target_, select _Library_.
+- Inside the project folder, create a new C++ implementation and header file. 
+- For this example we will call the files CxxTest, so we should have a CxxTest.cpp and CxxTest.hpp. Source code listed at the end of this section.
+- Change the "Terget Membership" setting, hpp file should set the target to the library target, and cpp file should be set to both the project and library target. 
```

Comment:
> cpp file should be set to both the project and library target

I think the cpp file should belong to a single target: the C++ library. Was there any issue that got fixed when you added the cpp file to the second target?